src/nginx/error.cc

```diff
@@ -207,8 +208,19 @@ ngx_int_t ngx_esp_error_body_filter(ngx_http_request_t *r, ngx_chain_t *in) {
         // Serialize error as JSON
         ngx_buf_t *body = nullptr;
         ngx_str_t json_error;
-        if (ngx_str_copy_from_std(r->pool, ctx->status.ToJson(), &json_error) !=
-            NGX_OK) {
+
+        if (ctx->grpc_status_details != nullptr) {
```

Comment:
Add comment
// if there is grpc-status-detail-bin response header, generate error json body with that header.

Reply:
Added.